### ISSUE TITLE
Failure during initialising environment on Node21 as navigator object is populated as an empty object

### ISSUE THREAD

**Issue by kimdanielarthur-cowlabs** [iamcal/js-emoji#180]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

navigator object is present in Node >21 according to node docs https://nodejs.org/en/blog/announcements/v21-release-announce#navigator-object-integration 

currently this is used to detect if we are running in a browser or not. A quick fix below, there may be better ways as i think navigator might have keys set in some scenarios

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/emoji-js/lib/emoji.js b/node_modules/emoji-js/lib/emoji.js
index 626a0fd..f9b1405 100644
--- a/node_modules/emoji-js/lib/emoji.js
+++ b/node_modules/emoji-js/lib/emoji.js
@@ -588,7 +588,7 @@
 		self.inits.env = 1;
 		self.replace_mode = 'img';
 		var supports_css = false;
-		if (typeof(navigator) !== 'undefined') {
+		if (typeof(navigator) !== 'undefined' && Object.keys(navigator).length > 0) {
 			var ua = navigator.userAgent;
 			if (typeof window !== 'undefined' && window.getComputedStyle){
 				try {
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by iamcal** [comment]
Does the current trunk version work for you? I'll cut a new release if it looks good

**Comment by kimdanielarthur-cowlabs** [comment]
Sorry for not replying! I moved on to another project and left my wokaround as a patch so have not tested.

**Comment by alexanderweiss** [comment]
Running into a similar issue node 22.11 (now LTS). Tried 6b99ad1bb537653e8c6f81d5185f0784fb4f1dc6 and it didn't fix the issue. Specifically, this seems to be about `navigator.product`

```
TypeError: Cannot read properties of undefined (reading 'match')
      at emoji.init_env (node_modules/emoji-js/lib/emoji.js:607:26)
      at new emoji (node_modules/emoji-js/lib/emoji.js:137:8)
```

**Comment by youyinnn** [comment]
``` javascript
...../node_modules/emoji-js/lib/emoji.js:607
                        if (navigator.product.match(/ReactNative/i)){
                                              ^

TypeError: Cannot read properties of undefined (reading 'match')
```

Solution?

**Comment by danielsiwiec** [comment]
same issue here...

**Comment by asaflanzer** [comment]
@iamcal Can you publish a version with this fix? 🙏

**Comment by iamcal** [comment]
I just published 3.8.1 - thanks for confirming the fix
